k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -137,22 +137,30 @@
     public static class UidReverseComparator implements Comparator<Message> {
         @Override
         public int compare(Message o1, Message o2) {
-            if (o1 == null || o2 == null || o1.getUid() == null || o2.getUid() == null) {
-                return 0;
-            }
-            int id1, id2;
+            Integer id1, id2;
```

Comment:
I'd rename `o1` and `o2` to `messageLeft` and `messageRight`. `id1` and `id2` to `uidLeft` and `uidRight`.